tests/test_connector.py

```diff
@@ -4454,3 +4455,29 @@ async def test_available_connections_no_limits(
     connection1.close()
     assert conn._available_connections(key) == 1
     assert conn._available_connections(other_host_key2) == 1
+
+
+async def test_connect_tunnel_connection_release(
+    loop: asyncio.AbstractEventLoop,
+) -> None:
+    """Test _ConnectTunnelConnection.release() does not pool the connection."""
+    connector = mock.Mock()
+    key = mock.Mock()
+    protocol = mock.Mock()
```

Comment:
Will autospec work here?
```suggestion
    connector = mock.create_autospec(BaseConnector, spec_set=True, instance=True)
    key = mock.create_autospec(ConnectionKey, spec_set=True, instance=True)
    protocol = mock.create_autospec(ResponseHandler, spec_set=True, instance=True)
```

I can play with this if you want to jump on that other issue..

Reply:
You beat me to it..